### PR TITLE
DPMMA-2550 Point Groups API Docs

### PR DESCRIPTION
**Pull request by patrykgruszka**
Docs for https://github.com/mautic/mautic/pull/13517

### CODE REVIEW

File: docs/rest_api/contacts.rst [hunk 1]
```diff
@@ -1055,6 +1055,316 @@ To subtract Points from a Contact and return a 404 if the Contact isn't found:
        "success": true
    }
 
+.. vale off
+
+Get All Contact Point Groups Scores
```
Comment (RCheesley) [inline docs/rest_api/contacts.rst]:
```suggestion
Get all Contact Point Groups scores
```
We only need to capitalise the first word and any Mautic-specific terms

File: docs/rest_api/contacts.rst [hunk 2]
```diff
@@ -1055,6 +1055,316 @@ To subtract Points from a Contact and return a 404 if the Contact isn't found:
...
+***********************************
+
+.. vale on
+
+Get all Point Group scores for a specific contact.
```
Comment (RCheesley) [inline docs/rest_api/contacts.rst]:
```suggestion
Get all Point Group scores for a specific Contact.
```

File: docs/rest_api/contacts.rst [hunk 3]
```diff
@@ -1055,6 +1055,316 @@ To subtract Points from a Contact and return a 404 if the Contact isn't found:
...
+                "score": 21,
+                "group": {
+                    "id": 6,
+                    "name": "A"
+                }
+            }
+        ]
+    }
+
+The response will contain a list of Point Group scores for the Contact.
```
Comment (RCheesley) [inline docs/rest_api/contacts.rst]:
```suggestion
The response contains a list of Point Group scores for the Contact.
```

Vale fix

File: docs/rest_api/contacts.rst [hunk 4]
```diff
@@ -1055,6 +1055,316 @@ To subtract Points from a Contact and return a 404 if the Contact isn't found:
...
+
+.. vale off
+
+Get Single Contact Point Group Score
```
Comment (RCheesley) [inline docs/rest_api/contacts.rst]:
```suggestion
Get single Contact Point Group score
```

File: docs/rest_api/contacts.rst [hunk 5]
```diff
@@ -1055,6 +1055,316 @@ To subtract Points from a Contact and return a 404 if the Contact isn't found:
...
+************************************
+
+.. vale on
+
+Get the score for a specific Point Group for a contact.
```
Comment (RCheesley) [inline docs/rest_api/contacts.rst]:
```suggestion
Get the score for a specific Point Group for a Contact.
```

File: docs/rest_api/contacts.rst [hunk 6]
```diff
@@ -1055,6 +1055,316 @@ To subtract Points from a Contact and return a 404 if the Contact isn't found:
...
+        "groupScore": {
+            "score": 21,
+            "group": {
+                "id": 6,
+                "name": "A"
+            }
+        }
+    }
+
+The response will contain the score for the specified Point Group for the Contact.
```
Comment (RCheesley) [inline docs/rest_api/contacts.rst]:
```suggestion
The response contains the score for the specified Point Group for the Contact.
```
Vale fix

File: docs/rest_api/contacts.rst [hunk 7]
```diff
@@ -1055,6 +1055,316 @@ To subtract Points from a Contact and return a 404 if the Contact isn't found:
...
+
+.. vale off
+
+Add Contact Point Group Score
+*****************************
+
+.. vale on
+
+Add points to a specific Point Group for a contact.
```
Comment (RCheesley) [inline docs/rest_api/contacts.rst]:
```suggestion
Add Points to a specific Point Group for a Contact.
```

File: docs/rest_api/contacts.rst [hunk 8]
```diff
@@ -1055,6 +1055,316 @@ To subtract Points from a Contact and return a 404 if the Contact isn't found:
...
+        "groupScore": {
+            "score": 21,
+            "group": {
+                "id": 6,
+                "name": "A"
+            }
+        }
+    }
+
+The response will indicate the success of adding points to the specified Point Group for the Contact.
```
Comment (RCheesley) [inline docs/rest_api/contacts.rst]:
```suggestion
The response indicates the success of adding Points to the specified Point Group for the Contact.
```

Vale fix

File: docs/rest_api/contacts.rst [hunk 9]
```diff
@@ -1055,6 +1055,316 @@ To subtract Points from a Contact and return a 404 if the Contact isn't found:
...
+
+.. vale off
+
+Subtract Contact Point Group Score
```
Comment (RCheesley) [inline docs/rest_api/contacts.rst]:
```suggestion
Subtract Contact Point Group score
```

File: docs/rest_api/contacts.rst [hunk 10]
```diff
@@ -1055,6 +1055,316 @@ To subtract Points from a Contact and return a 404 if the Contact isn't found:
...
+**********************************
+
+.. vale on
+
+Subtract points from a specific Point Group for a contact.
```
Comment (RCheesley) [inline docs/rest_api/contacts.rst]:
```suggestion
Subtract Points from a specific Point Group for a Contact.
```

File: docs/rest_api/contacts.rst [hunk 11]
```diff
@@ -1055,6 +1055,316 @@ To subtract Points from a Contact and return a 404 if the Contact isn't found:
...
+        "groupScore": {
+            "score": 21,
+            "group": {
+                "id": 6,
+                "name": "A"
+            }
+        }
+    }
+
+The response will indicate the success of subtracting points from the specified Point Group for the Contact.
```
Comment (RCheesley) [inline docs/rest_api/contacts.rst]:
```suggestion
The response indicates the success of subtracting Points from the specified Point Group for the Contact.
```

File: docs/rest_api/contacts.rst [hunk 12]
```diff
@@ -1055,6 +1055,316 @@ To subtract Points from a Contact and return a 404 if the Contact isn't found:
...
+
+.. vale off
+
+Set Contact Point Group Score
+*****************************
+
+.. vale on
+
+Set the points for a specific Point Group for a contact.
```
Comment (RCheesley) [inline docs/rest_api/contacts.rst]:
```suggestion
Set the Points for a specific Point Group for a Contact.
```

File: docs/rest_api/contacts.rst [hunk 13]
```diff
@@ -1055,6 +1055,316 @@ To subtract Points from a Contact and return a 404 if the Contact isn't found:
...
+        "groupScore": {
+            "score": 21,
+            "group": {
+                "id": 6,
+                "name": "A"
+            }
+        }
+    }
+
+The response will indicate the success of setting the points for the specified Point Group for the Contact.
```
Comment (RCheesley) [inline docs/rest_api/contacts.rst]:
```suggestion
The response indicates the success of setting the Points for the specified Point Group for the Contact.
```

File: docs/rest_api/contacts.rst [hunk 14]
```diff
@@ -1055,6 +1055,316 @@ To subtract Points from a Contact and return a 404 if the Contact isn't found:
...
+
+.. vale off
+
+Divide Contact Point Group Score
```
Comment (RCheesley) [inline docs/rest_api/contacts.rst]:
```suggestion
Divide Contact Point Group score
```

File: docs/rest_api/contacts.rst [hunk 15]
```diff
@@ -1055,6 +1055,316 @@ To subtract Points from a Contact and return a 404 if the Contact isn't found:
...
+********************************
+
+.. vale on
+
+Divide the points of a specific Point Group for a contact by a value.
```
Comment (RCheesley) [inline docs/rest_api/contacts.rst]:
```suggestion
Divide the Points of a specific Point Group for a Contact by a value.
```

File: docs/rest_api/contacts.rst [hunk 16]
```diff
@@ -1055,6 +1055,316 @@ To subtract Points from a Contact and return a 404 if the Contact isn't found:
...
+        "groupScore": {
+            "score": 21,
+            "group": {
+                "id": 6,
+                "name": "A"
+            }
+        }
+    }
+
+The response will indicate the success of dividing the points of the specified Point Group for the Contact by the specified value.
```
Comment (RCheesley) [inline docs/rest_api/contacts.rst]:
```suggestion
The response indicates the success of dividing the Points of the specified Point Group for the Contact by the specified value.
```

File: docs/rest_api/contacts.rst [hunk 17]
```diff
@@ -1055,6 +1055,316 @@ To subtract Points from a Contact and return a 404 if the Contact isn't found:
...
+
+.. vale off
+
+Multiply Contact Point Group Score
```
Comment (RCheesley) [inline docs/rest_api/contacts.rst]:
```suggestion
Multiply Contact Point Group score
```

File: docs/rest_api/contacts.rst [hunk 18]
```diff
@@ -1055,6 +1055,316 @@ To subtract Points from a Contact and return a 404 if the Contact isn't found:
...
+**********************************
+
+.. vale on
+
+Multiply the points of a specific Point Group for a contact by a value.
```
Comment (RCheesley) [inline docs/rest_api/contacts.rst]:
```suggestion
Multiply the Points of a specific Point Group for a Contact by a value.
```

File: docs/rest_api/contacts.rst [hunk 19]
```diff
@@ -1055,6 +1055,316 @@ To subtract Points from a Contact and return a 404 if the Contact isn't found:
...
+        "groupScore": {
+            "score": 21,
+            "group": {
+                "id": 6,
+                "name": "A"
+            }
+        }
+    }
+
+The response will indicate the success of multiplying the points of the specified Point Group for the Contact by the specified value.
```
Comment (RCheesley) [inline docs/rest_api/contacts.rst]:
```suggestion
The response indicates the success of multiplying the Points of the specified Point Group for the Contact by the specified value.
```

File: docs/rest_api/point_groups.rst [hunk 20]
```diff
@@ -0,0 +1,338 @@
...
+     - Point Group description
+   * - ``isPublished``
+     - ``boolean``
+     - Whether the Point Group is published
+   * - ``dateAdded``
+     - ``datetime``
+     - Date/time Point Group was created
+   * - ``createdBy``
+     - ``int``
+     - ID of the user that created the Point Group
```
Comment (RCheesley) [inline docs/rest_api/point_groups.rst]:
```suggestion
     - ID of the User that created the Point Group
```

File: docs/rest_api/point_groups.rst [hunk 21]
```diff
@@ -0,0 +1,338 @@
...
+   * - ``createdByUser``
+     - ``string``
+     - Name of the user that created the Point Group
```
Comment (RCheesley) [inline docs/rest_api/point_groups.rst]:
```suggestion
     - Name of the User that created the Point Group
```

File: docs/rest_api/point_groups.rst [hunk 22]
```diff
@@ -0,0 +1,338 @@
...
+   * - ``dateModified``
+     - ``datetime/null``
+     - Date/time Point Group was last modified
+   * - ``modifiedBy``
+     - ``int``
+     - ID of the user that last modified the Point Group
```
Comment (RCheesley) [inline docs/rest_api/point_groups.rst]:
```suggestion
     - ID of the User that last modified the Point Group
```

File: docs/rest_api/point_groups.rst [hunk 23]
```diff
@@ -0,0 +1,338 @@
...
+   * - ``modifiedByUser``
+     - ``string``
+     - Name of the user that last modified the Point Group
```
Comment (RCheesley) [inline docs/rest_api/point_groups.rst]:
```suggestion
     - Name of the User that last modified the Point Group
```

File: docs/rest_api/point_groups.rst [hunk 24]
```diff
@@ -0,0 +1,338 @@
...
+     - Point Group description
+   * - ``isPublished``
+     - ``boolean``
+     - Whether the Point Group is published
+   * - ``dateAdded``
+     - ``datetime``
+     - Date/time Point Group was created
+   * - ``createdBy``
+     - ``int``
+     - ID of the user that created the Point Group
```
Comment (RCheesley) [inline docs/rest_api/point_groups.rst]:
```suggestion
     - ID of the User that created the Point Group
```

File: docs/rest_api/point_groups.rst [hunk 25]
```diff
@@ -0,0 +1,338 @@
...
+   * - ``createdByUser``
+     - ``string``
+     - Name of the user that created the Point Group
```
Comment (RCheesley) [inline docs/rest_api/point_groups.rst]:
```suggestion
     - Name of the User that created the Point Group
```

File: docs/rest_api/point_groups.rst [hunk 26]
```diff
@@ -0,0 +1,338 @@
...
+   * - ``dateModified``
+     - ``datetime/null``
+     - Date/time Point Group was last modified
+   * - ``modifiedBy``
+     - ``int``
+     - ID of the user that last modified the Point Group
```
Comment (RCheesley) [inline docs/rest_api/point_groups.rst]:
```suggestion
     - ID of the User that last modified the Point Group
```

File: docs/rest_api/point_groups.rst [hunk 27]
```diff
@@ -0,0 +1,338 @@
...
+   * - ``modifiedByUser``
+     - ``string``
+     - Name of the user that last modified the Point Group
```
Comment (RCheesley) [inline docs/rest_api/point_groups.rst]:
```suggestion
     - Name of the User that last modified the Point Group
```

File: docs/rest_api/point_groups.rst [hunk 28]
```diff
@@ -0,0 +1,338 @@
...
+   <?php
+
+   $data = [
+       'name'        => 'Group A',
+       'description' => 'This is my first Point Group created via API.'
+   ];
+
+   $pointGroup = $pointGroupApi->create($data);
+
+Create a new Point Group.
```
Comment (RCheesley) [inline docs/rest_api/point_groups.rst]: Is this meant to be a header, or maybe it's extra text that isn't needed? It wasn't clear to me.

File: docs/rest_api/point_groups.rst [hunk 29]
```diff
@@ -0,0 +1,338 @@
...
+
+   $id   = 1;
+   $data = [
+       'name'        => 'New Point Group name',
+       'description' => 'Updated description of the Point Group.'
+   ];
+
+   $pointGroup = $pointGroupApi->edit($id, $data);
+
+Edit a Point Group.
```
Comment (RCheesley) [inline docs/rest_api/point_groups.rst]: Same as before - wasn't sure what this was for?

**Review by RCheesley — CHANGES_REQUESTED** [review]
Thanks for the PR @patrykgruszka - I have reviewed and suggested some fixes for the Vale warnings relating to active v passive voice, and also a few capitals missing. 

There were a couple of one-liners which I wasn't sure if they were there on purpose - if so then that is fine!

File: docs/rest_api/point_groups.rst (hunk 28 above)
Comment (patrykgruszka) [reply docs/rest_api/point_groups.rst]: It doesn't add any value, let's remove it

**Comment by patrykgruszka** [comment]
@RCheesley thanks! I've added your suggestions.

**Review by RCheesley — APPROVED** [review]
Good with me, I've tagged the Edu Team for a final review and merge! Thanks @patrykgruszka 🚀

**Comment by patrykgruszka** [comment]
Great! Also https://github.com/mautic/mautic/pull/13517 needs to be merged first.

**Comment by RCheesley** [comment]
Great - thanks for the heads up I've added the label to that effect. Might not have time to get it in for 5.1 tomorrow but maybe, if there are some keen testers/reviewers today!

**Pull request merged**